src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -251,9 +251,19 @@ public boolean perform(AbstractBuild<?, ?> build,
                     if (t.getTargetRepoName() == null)
                         throw new AbortException("No target repo to push to defined");
 
-                    final String tagName = environment.expand(t.getTagName());
-                    final String tagMessage = hudson.Util.fixNull(environment.expand(t.getTagMessage()));
-                    final String targetRepo = environment.expand(t.getTargetRepoName());
+                    String tagName;
+                    String tagMessage;
+                    String targetRepo;
+                    try {
+                        Class.forName("org.jenkinsci.plugins.tokenmacro.TokenMacro");
```

Comment:
AIUI this will still cause issues if the token macro plugin is not installed. we should just either mandate as mandatory dependency or else we need to play dancing games with classloading to include it correctly. in either case this implementation is not correct as it currently stands